argobots.spec

```diff
@@ -1,22 +1,27 @@
 Name: argobots
-Version: 1.0
-Release: 1%{?dist}
+Version: 1.1~a1
+Release: 1%{?dist}.g7fd1987
 Summary: Lightweight, low-level threading and tasking framework
 Group: System Environment/Libraries
 License: UChicago Argonne, LLC -- Argobots License
 Url: http://www.argobots.org/
 #Source: https://api.github.com/repos/pmodels/$(NAME)/tarball/31703b1
 Source: https://github.com/pmodels/%{name}/releases/download/v%{version}/%{name}-%{version}.tar.gz
```

Comment:
Because you are now fetching a pre-release version which RPM puts a `~` in, but upstream does not, you cannot use `%{version}` here directly, as it as the `~` in it.

@mjean308 did some packaging work last week that handles this case, and it would be a good fit here.  The changes she made are in this diff: https://github.com/daos-stack/hdf5/pull/20/files#diff-3b7cf60b9b14af8ded744b306cb4b75cc842d2db5e782edf5fff11353db5e81c.  When applied, they future-proof the package against being either a non-pre-release or a pre-release and don't run into these problems.  We will be migrating the technique she used as standard in our specfiles, so adding it here would be a step in that direction.

So generally speaking you want to do something like this at the top of this file:

```
%global major 1
%global minor 13
%global bugfix 0
%global prerelease a1
%global upstream_version %{major}.%{minor}.%{?bugfix:.%{bugfix}}%{?prerelease}
```

This defines the version components each time the version needs updating, and then if you make this change to the `Source:` line above:

```suggestion
Source: https://github.com/pmodels/%{name}/releases/download/v%{upstream_version}/%{name}-%{upstream-version}.tar.gz
```

We should never need to update the Source: line between full and pre-releases and just update the `%global` variables at the top of the file.

Reply:
Humm, ok for these spec file enhancements, but the root problem is elsewhere as the correct link to get the source tarball is "https://github.com/pmodels/argobots/releases/download/v1.1a1/argobots-1.1a1.tar.gz", so my original %version naming of "1.1a1" should have worked !!!

Reply:
> my original %version naming of "1.1a1" should have worked !!!

Except that's it's an illegal version for RPM.  The `~` is required prior to the `%{prerelease}` so that RPM understands that this is a _pre-release_ of `1.1` and that when a final `1.1` package is available it will update a host with `1.1~a1` installed.  If the `1.1a1` release is versioned like that in the RPM, then rpm will consider that newer (because it's lexicographically greater than `1.1`) and never upgrade `1.1a1` to `1.1`.

So the goal here is to make the `Version:` correct and make the download name be the transformation of the `Version:` value, which your latest update to packaging/` should do.  It removes any `~` from `%{version}`.  And it is working.  I tested  this PR locally.